sensor_control/vision/symbols/src/FrameProc.cpp

```diff
@@ -0,0 +1,110 @@
+#include "FrameProc.h"
+
+int FrameProc::blur_size = 7;
+FrameProc::FrameProc()
+{
+  red = ColorThresh{Scalar(0,10, 100), Scalar(30, 255, 255)};
+	red2 = ColorThresh{Scalar(155, 10, 100), Scalar(180, 255, 255)};
+	blue = ColorThresh{Scalar(90,100,100),Scalar(150,255,255)};
+	green = ColorThresh{Scalar(30, 85, 25),Scalar(85,255,255)};
+
+  #ifdef DO_DEBUG
+  namedWindow("blured",CV_WINDOW_AUTOSIZE);
+  //namedWindow("hsv",CV_WINDOW_AUTOSIZE);
+  namedWindow("blue",CV_WINDOW_AUTOSIZE);
+  namedWindow("green",CV_WINDOW_AUTOSIZE);
+  namedWindow("red",CV_WINDOW_AUTOSIZE);
+  createTrackbar("Blur Size","blured",&blur_size,100);
+  #endif
+}
+void FrameProc::init(ros::NodeHandle& nh)
+{
+  //Set HSV values
+  nh.getParam("hsv/red1/low/H",red.low[0]);
```

Comment:
Consider [retrieving a list](http://wiki.ros.org/roscpp/Overview/Parameter%20Server#Retrieving_Lists) from the ros parameter server instead of individually like this - it would save a lot of space.